src/NHibernate/Hql/Ast/ANTLR/Tree/FromElement.cs

```diff
@@ -337,9 +337,10 @@ public string RenderIdentifierSelect(int size, int k)
 		/// <returns>the property select SQL fragment.</returns>
 		public string RenderPropertySelect(int size, int k)
 		{
-			return IsAllPropertyFetch
-				? _elementType.RenderPropertySelect(size, k, IsAllPropertyFetch)
-				: _elementType.RenderPropertySelect(size, k, _fetchLazyProperties);
+			return _elementType.RenderPropertySelect(
+				size,
+				k,
+				IsAllPropertyFetch || _fetchLazyProperties?.Contains(TableAlias) == true);
```

Comment:
Didn't dig in to changes but on a first look it looks wrong.  Table alias and property names? I see no connection..

Reply:
`TableAlias` is a parameter which will go to `PropertySelectFragment` as "name". I just moved the decision to the caller.

Reply:
You're right. This is just nonsense.

Reply:
I've mistaken `name` for the "property" name, but it is a table name/alias.